main.cpp

```diff
@@ -158,7 +159,20 @@ int main(int argc, char *argv[])
 
     // Perform safe shutdown
     signal(SIGINT, terminationSignalHandler);
-    QObject::connect(&a, &QCoreApplication::aboutToQuit, &mavsdkVehicleServer, &MavsdkVehicleServer::saveParametersToXmlFile);
+    QObject::connect(&a, &QCoreApplication::aboutToQuit, [&](){
+        mUbloxRover->saveOnShutdown();
+        mavsdkVehicleServer.saveParametersToXmlFile();
+    });
+    QObject::connect(&mavsdkVehicleServer, &MavsdkVehicleServer::shutdownOrRebootOnboardComputer, [&](int value){
+        qApp->quit();
+        if (value == 1) {
+           qDebug() << "\nSystem reboot...";
+           QProcess::startDetached("sudo", QStringList() << "shutdown" << "-r" << "now");
+        }else if (value == 2){
```

Comment:
Should be bool or enum (see https://github.com/RISE-Dependable-Transport-Systems/WayWise/pull/30/files#r1115310881).

Reply:
Fixed